crates/driver/src/domain/competition/settled/mod.rs

```diff
@@ -0,0 +1,268 @@
+use {
+    super::{
+        auction,
+        order::{self, Side},
+    },
+    crate::{domain::eth, util::conv::u256::U256Ext},
+    number::conversions::big_rational_to_u256,
+    std::collections::HashMap,
+};
+
+/// Settlement built from onchain calldata.
+#[derive(Debug, Clone)]
+pub struct Settlement {
+    trades: Vec<Trade>,
+}
+
+impl Settlement {
+    pub fn new(trades: Vec<Trade>) -> Self {
+        Self { trades }
+    }
+
+    /// Score of a settlement as per CIP38
+    ///
+    /// Denominated in NATIVE token
+    pub fn score(
+        &self,
+        prices: &HashMap<eth::TokenAddress, auction::NormalizedPrice>,
+    ) -> Option<eth::TokenAmount> {
```

Comment:
Should this rather return a `Result`? If the `score` cannot be computed because prices are missing this is a significant problem, right?
Same thought for other functions as well.

Reply:
Added error handling.